gcc/config/sol2.h

```diff
@@ -265,8 +265,7 @@ along with GCC; see the file COPYING3.  If not see
 
 #undef LINK_ARCH32_SPEC_BASE
 #define LINK_ARCH32_SPEC_BASE \
-  "%{G:-G} \
-   %{YP,*} \
+  "%{YP,*} \
```

Comment:
I'm not quite following this change.

Reply:
I'm just relocating things a bit to match `-shared`. The following is from before this change:

```
*link:
%{h*} %{v:-V}    %{!shared:%{!static:%{rdynamic: }}}    %{static:-dn -Bstatic}    %{shared:-G -dy %{!mimpure-text:-z text}} %{shared|shared-libgcc:-M %slibgcc-unwind.map} %{mclear-hwcap|fopenmp*:-M %sclearcap.map}    %{symbolic:-Bsymbolic -G -dy -z text}    %(link_arch)    %{Qy:} %{!Qn:-Qy}
```

This is the link command, which adds `-G` if the original command line included `-shared`. Towards the end, it includes `%{link_arch}`

```
*link_arch:
 %{m32:%(link_arch32)} %{m64:%(link_arch64)} %{!m32:%{!m64:%(link_arch_default)}}
```

which includes one of the following blocks, all of which also include `%{G:-G}` 

```
*link_arch32:
%{G:-G}    %{YP,*}    %{R*}    %{!YP,*:%{p|pg:-Y P,%R/usr/gcc/8/lib:%R/lib:%R/usr/lib -R %R/usr/gcc/8/lib -L %R/usr/gcc/8/lib}     %{!p:%{!pg:-Y P,%R/usr/gcc/8/lib:%R/lib:%R/usr/lib -R %R/usr/gcc/8/lib -L %R/usr/gcc/8/lib}}}

*link_arch64:
%{mcmodel=medlow:-M /usr/lib/ld/amd64/map.below4G} %{G:-G}    %{YP,*}    %{R*}    %{!YP,*:%{p|pg:-Y P,%R/usr/gcc/8/lib/amd64:%R/lib/amd64:%R/usr/lib/amd64 -R %R/usr/gcc/8/lib/amd64 -L %R/usr/gcc/8/lib/amd64}            %{!p:%{!pg:-Y P,%R/usr/gcc/8/lib/amd64:%R/lib/amd64:%R/usr/lib/amd64 -R %R/usr/gcc/8/lib/amd64 -L %R/usr/gcc/8/lib/amd64}}}

*link_arch_default:
%{G:-G}    %{YP,*}    %{R*}    %{!YP,*:%{p|pg:-Y P,%R/usr/gcc/8/lib:%R/lib:%R/usr/lib -R %R/usr/gcc/8/lib -L %R/usr/gcc/8/lib}     %{!p:%{!pg:-Y P,%R/usr/gcc/8/lib:%R/lib:%R/usr/lib -R %R/usr/gcc/8/lib -L %R/usr/gcc/8/lib}}}
```

Since I've changed the `*link:` section so that `-G` is added for both `-shared` and `-G`, the first clause in each of these `link_arch` lines is no longer needed.